sql/core/src/test/resources/sql-tests/inputs/udf/pgSQL/udf-join.sql

```diff
@@ -0,0 +1,2081 @@
+CREATE TABLE t2 (name STRING, n INTEGER) USING parquet;
+CREATE TABLE t3 (name STRING, n INTEGER) USING parquet;
+
+INSERT INTO t1 VALUES ( 'bb', 11 );
+INSERT INTO t2 VALUES ( 'bb', 12 );
+INSERT INTO t2 VALUES ( 'cc', 22 );
+INSERT INTO t2 VALUES ( 'ee', 42 );
+INSERT INTO t3 VALUES ( 'bb', 13 );
+INSERT INTO t3 VALUES ( 'cc', 23 );
+INSERT INTO t3 VALUES ( 'dd', 33 );
+
+SELECT * FROM t1 FULL JOIN t2 USING (name) FULL JOIN t3 USING (name);
+
+--
+-- Test interactions of join syntax and subqueries
+--
+
+-- Basic cases (we expect planner to pull up the subquery here)
+SELECT * FROM
+(SELECT udf(name) as name, t2.n FROM t2) as s2
+INNER JOIN
+(SELECT udf(udf(name)) as name, t3.n FROM t3) s3
+USING (name);
+
+SELECT * FROM
+(SELECT udf(udf(name)) as name, t2.n FROM t2) as s2
+LEFT JOIN
+(SELECT udf(name) as name, t3.n FROM t3) s3
+USING (name);
+
+SELECT udf(name), udf(udf(s2.n)), udf(s3.n) FROM
+(SELECT * FROM t2) as s2
+FULL JOIN
+(SELECT * FROM t3) s3
+USING (name);
+
+-- Cases with non-nullable expressions in subquery results;
+-- make sure these go to null as expected
+SELECT * FROM
+(SELECT udf(udf(name)) as name, udf(n) as s2_n, udf(2) as s2_2 FROM t2) as s2
+NATURAL INNER JOIN
+(SELECT udf(name) as name, udf(udf(n)) as s3_n, udf(3) as s3_2 FROM t3) s3;
+
+SELECT * FROM
+(SELECT udf(name) as name, udf(udf(n)) as s2_n, 2 as s2_2 FROM t2) as s2
+NATURAL LEFT JOIN
+(SELECT udf(udf(name)) as name, udf(n) as s3_n, 3 as s3_2 FROM t3) s3;
+
+SELECT * FROM
+(SELECT udf(name) as name, udf(n) as s2_n, 2 as s2_2 FROM t2) as s2
+NATURAL FULL JOIN
+(SELECT udf(udf(name)) as name, udf(udf(n)) as s3_n, 3 as s3_2 FROM t3) s3;
+
+SELECT * FROM
+(SELECT udf(udf(name)) as name, udf(n) as s1_n, 1 as s1_1 FROM t1) as s1
+NATURAL INNER JOIN
+(SELECT udf(name) as name, udf(n) as s2_n, 2 as s2_2 FROM t2) as s2
+NATURAL INNER JOIN
+(SELECT udf(udf(udf(name))) as name, udf(n) as s3_n, 3 as s3_2 FROM t3) s3;
+
+SELECT * FROM
+(SELECT udf(name) as name, udf(n) as s1_n, udf(udf(1)) as s1_1 FROM t1) as s1
+NATURAL FULL JOIN
+(SELECT udf(name) as name, udf(udf(n)) as s2_n, udf(2) as s2_2 FROM t2) as s2
+NATURAL FULL JOIN
+(SELECT udf(udf(name)) as name, udf(n) as s3_n, udf(3) as s3_2 FROM t3) s3;
+
+SELECT name, udf(udf(s1_n)), udf(s2_n), udf(s3_n) FROM
+(SELECT name, udf(udf(n)) as s1_n FROM t1) as s1
+NATURAL FULL JOIN
+  (SELECT * FROM
+    (SELECT name, udf(n) as s2_n FROM t2) as s2
+    NATURAL FULL JOIN
+    (SELECT name, udf(udf(n)) as s3_n FROM t3) as s3
+  ) ss2;
+
+SELECT * FROM
+(SELECT name, n as s1_n FROM t1) as s1
+NATURAL FULL JOIN
+  (SELECT * FROM
+    (SELECT name, udf(udf(n)) as s2_n, 2 as s2_2 FROM t2) as s2
+    NATURAL FULL JOIN
+    (SELECT name, udf(n) as s3_n FROM t3) as s3
+  ) ss2;
+
+-- Constants as join keys can also be problematic
+SELECT s1.name, udf(s1_n), s2.name, udf(udf(s2_n)) FROM
+  (SELECT name, udf(n) as s1_n FROM t1) as s1
+FULL JOIN
+  (SELECT name, udf(2) as s2_n FROM t2) as s2
```

Comment:
The purpose of this test, looks like, is to use constants as join key, so I think we should not rewrite `2` to `udf(2)`?